File: Java/Day02Assignment/MyOwnAutoShopProject/Truck.java
package com.sonata;

public class Truck extends Car{
	int weight;
	public double getSalePrice(int weight) {
		double total=super.getSalePrice()*0.8;
		if(weight>2000) {
			total=super.getSalePrice()*0.9;
		}
		return total;
	}
	
	public static void main(String args[]) {
		Truck t1= new Truck();
		t1.speed=300;
		t1.regularPrice=1000000;
		t1.color="red";
		
		t1.display();
		System.out.println(t1.getSalePrice(2050));
	}

}

File: TypeScript/Truck.ts
import { Car } from "./Car";

class Truck extends Car{
    weight:number;
    getSalePrice(): number {
        if(this.weight>2000){
            return this.regularPrice=this.regularPrice*0.9;
        }
        else{
            return this.regularPrice=this.regularPrice*0.8;
        }
    }
    display(){
        console.log(this.speed);
        console.log(this.color);
        console.log(this.weight);
        console.log(this.regularPrice)
        console.log(this.getSalePrice());
    }
}
var t=new Truck();
t.speed=80;
t.color="red";
t.weight=2000;
t.regularPrice=40000;
t.display();